source/fundamentals/typescript.txt

```diff
@@ -142,6 +142,127 @@ section.
 
 .. _node-driver-limitations:
 
+
+Insert Operations and the _id Field
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+How you specify the ``_id`` field in type parameters passed to your
+``Collection`` instance affects the behavior
+of insert operations. The following table describes how different
+``_id`` field specifications affect insert operations:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 20 20 20 40
+
+   * - ``_id`` field type
+     - Example Type
+     - Required on insert
+     - Behavior on insert
+
+   * - | Unspecified
+     - | Not applicable
+     - | No
+     - | The driver creates an 
+         :manual:`ObjectId </reference/method/ObjectId/>`
+         value for each inserted document.
+
+   * - | Specified
+     - | ``{ _id: number };`` 
+     - | Yes
+     - | If you do not specify a value for the ``_id`` field in an insert operation,
+         the driver raises an error.
+
+   * - | Specified as optional
+     - | ``{ _id?: number };``
+     - | No
+     - | If you do not specify the ``_id`` field in an insert operation,
+         the driver adds an ``_id`` field value generated by a
+         **primary key factory**.
+       | A primary key factory is a class that defines a system for creating
+         unique identifiers for documents and implements the ``PkFactory``
+         interface.
+       | The default primary key factory generates ``_id`` values of type
+         ``ObjectId``.
+
+.. important:: Data Consistency and Optional _id Types
+
+   If you do not specify a primary key factory method that generates
+   ``_id`` values of your desired type, you may experience data inconsistency.
+   This occurs because the driver's default primary factory method generates
+   ``_id`` values of type ``ObjectId`` regardless of the type you specify for the 
+   ``_id`` field in your type parameter.
+
+   To show this behavior, assume you apply the following interface to
+   your ``Collection`` instance:
+
+   .. code-block:: typescript
+
+      interface I {
+        _id: number;
+        name: string;
+      } 
+
+   The following sequence of insert operations inserts one document
+   with an ``_id`` of type ``number`` and one document with an 
+   ``_id`` of type ``ObjectId`` into your collection:
+
+   .. code-block:: typescript
+
+      collection.insertOne({ _id: 1, name: "Scar" });
+      collection.insertOne({ name: "Ursula" });
```

Comment:
question: Won't this trigger an error since the interface makes `_id` required? If not, is the data in the table on lines 154-186 correct?

Reply:
You are correct, I missed a question mark on line 202.